### PR TITLE
Splunk HEC add sslContextParameters

### PR DESCRIPTION
**Pull request by JiriOndrusek**
second (and the last) part of fix for https://issues.apache.org/jira/browse/CAMEL-20393

PR adds sslConfiguratiionParameters int SplunkHec component.

There are no test of this feature in the component. I'm currently working on containerized tests of ssl in camel-quarkus (https://github.com/apache/camel-quarkus/pull/6302) Once I'm done with those tests, I'd  like to add some test (perhaps mocked ones) for ssl into Camel as well.

# Description

# Target

- [ ] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [ ] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [ ] I checked that each commit in the pull request has a meaningful subject line and body.

- [ ] I have run `mvn clean install -DskipTests` locally and I have committed all auto-generated changes

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Review by oscerd — APPROVED** [review]

**Review by davsclaus — APPROVED** [review]

File: components/camel-splunk-hec/src/main/java/org/apache/camel/component/splunkhec/SplunkHECProducer.java [hunk 1]
```diff
@@ -68,8 +68,16 @@ protected void doStart() throws Exception {
 
             connManager = new PoolingHttpClientConnectionManager(registryBuilder.build());
         } else {
-            connManager = new PoolingHttpClientConnectionManager();
+            //            connManager = new PoolingHttpClientConnectionManager();
```
Comment (davsclaus) [inline components/camel-splunk-hec/src/main/java/org/apache/camel/component/splunkhec/SplunkHECProducer.java:71]: Can you remove this old code

Comment (JiriOndrusek) [reply components/camel-splunk-hec/src/main/java/org/apache/camel/component/splunkhec/SplunkHECProducer.java:71]: sure, I'm sorry I missed that

Comment (JiriOndrusek) [reply components/camel-splunk-hec/src/main/java/org/apache/camel/component/splunkhec/SplunkHECProducer.java]: fixed

**Pull request merged**